pkg/registry/pod/strategy.go

```diff
@@ -464,12 +464,12 @@ func streamLocation(
 		return nil, nil, err
 	}
 	loc := &url.URL{
-		Scheme:   nodeScheme,
-		Host:     fmt.Sprintf("%s:%d", nodeHost, nodePort),
+		Scheme:   connectionInfo.Scheme,
+		Host:     fmt.Sprintf("%s:%d", connectionInfo.Hostname, connectionInfo.Port),
```

Comment:
while you're in here, `net.JoinHostPort(connectionInfo.Hostname, connectionInfo.Port)`